game/factions/faction.py

```diff
@@ -137,12 +118,64 @@ class Faction:
     #: both will use it.
     unrestricted_satnav: bool = False
 
-    def has_access_to_unittype(self, unit_class: GroundUnitClass) -> bool:
-        for vehicle in itertools.chain(self.frontline_units, self.artillery_units):
-            if vehicle.unit_class is unit_class:
-                return True
+    # Save all accessible units for performance increase
+    _accessible_units: list[UnitType[Any]] = field(default_factory=list)
+
+    def __getitem__(self, item: str) -> Any:
+        return getattr(self, item)
+
+    @property
+    def accessible_units(self) -> Iterator[UnitType[Any]]:
+        if not self._accessible_units:
+            self._load_accessible_units()
+        yield from self._accessible_units
+
+    @property
+    def air_defenses(self) -> list[str]:
```

Comment:
this is because in the previous version air_defenses were the generator names. I had a legacy function which parsed generator names and matched them to the corresponding templates. As i migrated all generators already this is obsolete. I changed all from str to the corresponding unittype already :)